### ISSUE TITLE
docker issues

### ISSUE THREAD

**Issue by colemickens** [raphael/linux-4.1-samus#24]
Not sure if it has to do with how the kernel is configured.

Can someone running this kernel try the repro steps here? https://github.com/docker/docker/issues/15070

Takes less than a minute. I'm not sure what else would be causing these issues.

**Comment by raphael** [comment]
I'm using docker containers daily without issues. I do see the file mode change you describe but that's not preventing the system from creating new terminals.

**Comment by colemickens** [comment]
@raphael So you are seeing the mount options change on the host's /dev/pts but you are not getting failures on creating new terminals?

Thank you for checking, I appreciate it. Still stumped as to what's wrong on my system, but it sounds like it's not the kernel.

**Comment by raphael** [comment]
Just to confirm:
```
➜  ~ mount | grep pts  
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,gid=5,mode=620)
➜  ~ docker run -a stdin -a stdout -i -t -v /home/raphael/src/linux-4.1-samus/build/debian/:/tmp/debian -v /boot:/boot ubuntu /bin/bash
root@33d51a13a93b:/# exit
exit
➜  ~ mount | grep pts  
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,mode=600)
```
but then I can still create new terminals.

**Comment by colemickens** [comment]
Very interesting, I wasn't expecting that; I'll have to look into this more tomorrow. Thanks again for the data points. Closing since this isn't an issue on your patches/kernel-config.

**Comment by raphael** [comment]
ok so I'm using terminology as my main terminal, I just tried with gnome-terminal and I'm getting the same error with it.

**Comment by colemickens** [comment]
I'll try booting the stock kernel with a mouse/keyboard and see what happens...

**Comment by colemickens** [comment]
Vanilla kernel does not cause a remount and thus does not have this issue.

It does look like it's an issue with the patches or with the kernel config, but I'm not sure which.

(As a sidenote, my touchpad and touchscreen work with the vanilla kernel. Audio output doesn't (only shows Digital options). Mic is appearing in GNOME, but `pavucontrol` doesn't see it.)

**Comment by colemickens** [comment]
```
$ uname -a
Linux pixel 4.1.3-8ph #1 SMP Thu Jul 23 18:23:50 PDT 2015 x86_64 GNU/Linux

$ ~/tmp/docker/contib/check-config.sh
info: reading kernel config from /proc/config.gz ...

Generally Necessary:
- cgroup hierarchy: properly mounted [/sys/fs/cgroup]
- CONFIG_NAMESPACES: enabled
- CONFIG_NET_NS: enabled
- CONFIG_PID_NS: enabled
- CONFIG_IPC_NS: enabled
- CONFIG_UTS_NS: enabled
- CONFIG_DEVPTS_MULTIPLE_INSTANCES: missing
- CONFIG_CGROUPS: enabled
- CONFIG_CGROUP_CPUACCT: enabled
- CONFIG_CGROUP_DEVICE: enabled
- CONFIG_CGROUP_FREEZER: enabled
- CONFIG_CGROUP_SCHED: enabled
- CONFIG_CPUSETS: enabled
- CONFIG_MEMCG: enabled
- CONFIG_MACVLAN: enabled (as module)
- CONFIG_VETH: enabled (as module)
- CONFIG_BRIDGE: enabled (as module)
- CONFIG_BRIDGE_NETFILTER: enabled (as module)
- CONFIG_NF_NAT_IPV4: enabled (as module)
- CONFIG_IP_NF_FILTER: enabled (as module)
- CONFIG_IP_NF_TARGET_MASQUERADE: enabled (as module)
- CONFIG_NETFILTER_XT_MATCH_ADDRTYPE: enabled (as module)
- CONFIG_NETFILTER_XT_MATCH_CONNTRACK: enabled (as module)
- CONFIG_NF_NAT: enabled (as module)
- CONFIG_NF_NAT_NEEDED: enabled
- CONFIG_POSIX_MQUEUE: enabled

Optional Features:
- CONFIG_MEMCG_KMEM: missing
- CONFIG_MEMCG_SWAP: missing
- CONFIG_MEMCG_SWAP_ENABLED: missing
- CONFIG_BLK_CGROUP: enabled
- CONFIG_IOSCHED_CFQ: enabled
- CONFIG_CGROUP_PERF: missing
- CONFIG_CGROUP_HUGETLB: missing
- CONFIG_NET_CLS_CGROUP: enabled (as module)
- CONFIG_NETPRIO_CGROUP: missing
- CONFIG_CFS_BANDWIDTH: enabled
- CONFIG_FAIR_GROUP_SCHED: enabled
- CONFIG_RT_GROUP_SCHED: enabled
- CONFIG_EXT3_FS: missing
- CONFIG_EXT3_FS_XATTR: missing
- CONFIG_EXT3_FS_POSIX_ACL: missing
- CONFIG_EXT3_FS_SECURITY: missing
    (enable these ext3 configs if you are using ext3 as backing filesystem)
- CONFIG_EXT4_FS: enabled (as module)
- CONFIG_EXT4_FS_POSIX_ACL: enabled
- CONFIG_EXT4_FS_SECURITY: enabled
- Storage Drivers:
  - "aufs":
    - CONFIG_AUFS_FS: missing
  - "btrfs":
    - CONFIG_BTRFS_FS: enabled (as module)
  - "devicemapper":
    - CONFIG_BLK_DEV_DM: enabled (as module)
    - CONFIG_DM_THIN_PROVISIONING: enabled (as module)
  - "overlay":
    - CONFIG_OVERLAY_FS: enabled (as module)
  - "zfs":
    - /dev/zfs: missing
    - zfs command: missing
    - zpool command: missing
```

Looks like there are a number of things missing that docker expects, including `CONFIG_DEVPTS_MULTIPLE_INSTANCES`.

**Comment by raphael** [comment]
Nice, thanks for looking into it. I'll build release 4.1-10 with the missing options enabled.

**Comment by raphael** [comment]
Just released 4.1-10 which adds the missing options and upgrades to 4.1.4. Tested it and I'm able to open a gnome-terminal after running docker. The file mode for `/dev/pts` remains 620. Thanks for the help on that one!
